### PR TITLE
add ENDOOM support (from Chocolate Doom)

### PR DESCRIPTION
**Pull request by rfomin**
Add libtextscreen for future use in multiplayer gui and ENDOOM.

### CODE REVIEW

**Comment by rfomin** [comment]
Not sure about linear scaling:
endoom

File: textscreen/txt_sdl.c [hunk 1]
```diff
@@ -0,0 +1,1009 @@
...
+    {
+        for (x1=x; x1<x_end; ++x1)
+        {
+            UpdateCharacter(x1, y1);
+        }
+    }
+
+    SDL_UnlockSurface(screenbuffer);
+
+    SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "linear");
```
Comment (fabiangreffrath) [inline textscreen/txt_sdl.c:466]: I guess for scaling up of blocky DOS font and pixel art, "nearest" makes more sense here.

File: textscreen/txt_sdl.c [hunk 2]
```diff
@@ -0,0 +1,1009 @@
...
+
+    // Instead, we draw everything into an intermediate 8-bit surface
+    // the same dimensions as the screen. SDL then takes care of all the
+    // 8->32 bit (or whatever depth) color conversions for us.
+    screenbuffer = SDL_CreateRGBSurface(0,
+                                        TXT_SCREEN_W * font->w,
+                                        TXT_SCREEN_H * font->h,
+                                        8, 0, 0, 0, 0);
+
+    SDL_RenderSetLogicalSize(renderer, screenbuffer->w, screenbuffer->h);
```
Comment (fabiangreffrath) [inline textscreen/txt_sdl.c:317]: Comment, please! If there are other parts of this library that you had to change to achieve the current state, please comment them as well.

File: Source/m_menu.c [hunk 3]
```diff
@@ -3624,6 +3625,9 @@ setup_menu_t gen_settings2[] = { // General Settings screen2
   {"Default compatibility", S_CHOICE|S_LEVWARN, m_null, G_X,
    G_Y4 + general_comp*8, {"default_complevel"}, 0, NULL, default_compatibility_strings},
 
+  {"Show ENDOOM", S_YESNO, m_null, G_X,
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3628]: I only want to see ENDOOM for PWADs that provide their own lump, so please turn this into a choice of three: "on", "off", PWAD only".

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: textscreen/txt_sdl.c (hunk 1 above)
Comment (rfomin) [reply textscreen/txt_sdl.c:466]: I tried it and it looks bad. Technically we have the wrong aspect ratio, but scaling the font is tricky. I think 1:1 linear scaling works best.

Comment (fabiangreffrath) [reply textscreen/txt_sdl.c:466]: Okay, fine with me then.

**Comment by fabiangreffrath** [comment]
Looks pretty complete to me. Ready to merge?

**Comment by rfomin** [comment]
Yes, I think it's ready.

**Pull request merged**